src/coresyftools/coresyftools/argument_parser.py

```diff
@@ -0,0 +1,108 @@
+from datetime import date
+import logging
+
+from argparse import ArgumentParser
+from manifest import validate_manifest, InvalidManifestException, MANIFEST_SCHEMA
+
+
+class CoReSyFArgumentParser():
+
+    MANIFEST_FILE_NAME = 'manifest.json'
+
+    def __init__(self, manifest, args=None, logger=None):
+        self.logger = logger or logging.getLogger(__name__)
+        self.bindings = {}
+        self.inputs = []
+        self.outputs = []
+        self.options = []
+        self.arg_parser = ArgumentParser()
+        self.manifest = manifest
+        validate_manifest(self.manifest)
+        is_valid, errors = self._validate_manifest(self.manifest)
+        if not is_valid:
+            raise InvalidManifestException(errors)
+
+    def _get_manifest_schema(self):
+        return MANIFEST_SCHEMA
+
+    def _validate_manifest(self, manifest):
+        return (True, [])
+
+    def _from_xsd_type_(self, xsd_type):
+        if xsd_type == 'boolean':
+            def parse_boolean(value):
+                if value == 'true':
+                    return True
+                elif value == 'false':
+                    return False
+                elif value == 1:
+                    return True
+                elif value == 0:
+                    return False
+                else:
+                    raise ValueError('invalid literal for boolean: {}'.format(value))
+            return parse_boolean
+        elif xsd_type == 'int':
+            return int
+        elif xsd_type == 'float':
+            return float
+        elif xsd_type == 'date':
+            return date
+        elif xsd_type == 'string':
+            return str
+
+    def _parse_parameter_arg_(self, arg):
+        name = '--' + arg['identifier']
+        _type = self._from_xsd_type_(arg['parameterType'])
+        _help = arg['description']
+        kwargs = {}
+        if 'options' in arg:
+            kwargs['choices'] = arg['options']
+        if 'default' in arg:
+            kwargs['default'] = arg['default']
+        if 'required' in arg:
+            kwargs['required'] = arg['required']
+        self.arg_parser.add_argument(name, type=_type, help=_help, **kwargs)
+        self.logger.debug('Parsed %s parameter argument.', name)
+        if arg['parameterType'] == 'boolean':
```

Comment:
What is the difference, in terms of clearness, between `add_identifier_to_options_if_boolean` or 
```python
 if arg['type'] == 'boolean':
            self.options.append(name)
 ``` ?
I would say the second is even more clear, since in the first one can ask "what options?", "if boolean what?".

Still, there is no need for a docstring, I can put a comment explaining why directly in the code.

Comment added.

Reply:
Agreed.